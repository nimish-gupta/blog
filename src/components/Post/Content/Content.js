// @flow strict
import React from 'react';
import * as Styled from './Content.style';

type Props = {
	body: string,
	title: string,
};

const Content = ({ body, title }: Props) => (
	<Styled.Content>
		<Styled.Title>{title}</Styled.Title>
		<Styled.Body dangerouslySetInnerHTML={{ __html: body }} />
	</Styled.Content>
);

export default Content;
