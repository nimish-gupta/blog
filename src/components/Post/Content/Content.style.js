/* eslint-disable prefer-template */
import styled from 'styled-components';
import { math } from 'polished';

import {
	lineHeight,
	marginTop,
	marginBottom,
	padding,
	breakpointMd,
} from '../../../assets/styled-components/mixins';

export const Content = styled.div`
	max-width: ${(props) => props.theme.layout.postSingleWidth};
	padding: 0 15px;
	margin: 0 auto;
	${breakpointMd`
    padding: 0;
  `}
`;

export const Title = styled.h1`
	font-size: ${(props) => math(props.theme.typographic.baseFontSize + ' * 2')};
	max-width: ${(props) => props.theme.layout.postWidth};
	margin-left: auto;
	margin-right: auto;
	font-weight: 600;
	text-align: center;
	${lineHeight(1.65)};
	${marginTop(1)};
	${marginBottom(0)};
	${breakpointMd`
    font-size: ${(props) =>
			math(props.theme.typographic.baseFontSize + ' * 3')};
    ${lineHeight(2.25)};
    ${marginTop(2.25)};
    ${marginBottom(1.5)});
  `}
`;

export const Body = styled.div`
	& figure {
		${marginBottom(1)};

		& blockquote {
			font-style: italic;
			text-align: center;
			margin-top: 0;
			${padding(1, 0)};

			& p {
				max-width: ${(props) => props.theme.layout.postWidth};
				font-size: ${(props) =>
					math(props.theme.typographic.baseFontSize + ' * 1.6817')};
				margin-top: 0;
				${marginBottom(1)};
				${lineHeight(1.5)}
			}
		}
	}

	& a {
		text-decoration: underline;
	}

	& * {
		max-width: ${(props) => props.theme.layout.postWidth};
		margin-left: auto;
		margin-right: auto;
	}

	& img {
		max-width: 100%;
	}
	${breakpointMd`
      font-size: ${(props) =>
				math(props.theme.typographic.baseFontSize + ' * 1.125')};
      ${lineHeight(1.125)};
      ${marginBottom(1.125)};

      & p {
        font-size: ${(props) =>
					math(props.theme.typographic.baseFontSize + ' * 1.125')};
        ${lineHeight(1.125)};
        ${marginBottom(1.125)};
      }
    `}
`;
