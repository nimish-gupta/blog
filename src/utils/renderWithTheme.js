import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import theme from '../assets/styled-components/_variables';

export default function renderWithTheme(component) {
	return renderer.create(
		<ThemeProvider theme={theme}>{component}</ThemeProvider>
	);
}
