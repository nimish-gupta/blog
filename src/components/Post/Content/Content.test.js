// @flow strict
import React from 'react';
import renderWithTheme from '../../../utils/renderWithTheme';
import Content from './Content';

describe('Content', () => {
	it('renders correctly', () => {
		const props = {
			title: 'test',
			body: '<p>test</p>',
		};

		const tree = renderWithTheme(<Content {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
