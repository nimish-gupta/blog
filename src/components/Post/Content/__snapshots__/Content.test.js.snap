// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
<div
  className="sc-AxjAm fHGxrB"
>
  <h1
    className="sc-AxirZ gbFByu"
  >
    test
  </h1>
  <div
    className="sc-AxiKw fZfbcy"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</div>
`;
